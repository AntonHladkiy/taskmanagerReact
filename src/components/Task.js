import React,{useEffect} from 'react';
import {Link} from "react-router-dom";
import $ from 'jquery';

const Task = ( { task , removeTask , editTask , onChangeCheckBoxHandle } ) => {
    useEffect(() => {
        $(document).ready(function() {
            $(".hover").hover(
                function () {
                    $("[id=" + "task_" + this.id.toString() + "]").removeClass("d-none")
                },
                function () {
                    let id=this.id;
                    setTimeout(function () {
                        $("[id=" + "task_" + id.toString() + "]").addClass("d-none")
                    }, 1500);
                }
            );
    })})
    return (
    <span className="task" key={ task.id }>
       <input type={ "checkbox" } id={ task.id } checked={ task.checked } ischecked={ task.checked.toString () } onChange={ onChangeCheckBoxHandle }/>
       <Link to={ "/view" } className={ "text-dark hover mr-3 hover" } id={ task.id } onClick={ () => {
           editTask ( task )
        } }>{ task.title }</Link>
    <Link to={ "/edit" } className="d-none" id={ "task_" + task.id.toString () }>
    <button className="btn btn-danger mr-3  d-none" id={ "task_" + task.id.toString () } onClick={ () => {editTask ( task )} }>Edit</button>
    </Link>
    <button className="btn btn-warning  d-none" id={ "task_" + task.id.toString () } onClick={ () => removeTask ( task.id ) }>Remove</button>
    </span>
    );
}


export default Task;