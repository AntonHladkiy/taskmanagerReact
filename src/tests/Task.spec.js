import React from 'react';
import { shallow} from 'enzyme';
import Task from '../components/Task';
const notCompletedTask = {
    id:0,
    title:'123',
    description:'123',
    dueDate:'132',
    priority:'123',
    done:false,
    checked:true
};
const completedTask = {
    id:0,
    title:'123',
    description:'123',
    dueDate:'132',
    priority:'123',
    done:true,
    checked:true
};
describe('Task completed', () => {
    let wrapped = shallow(<Task task={completedTask}/>);
    it('should render the Task Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });
});
describe('Task not completed', () => {
    let wrapped = shallow(<Task task={notCompletedTask}/>);
    it('should render the Task Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });

});