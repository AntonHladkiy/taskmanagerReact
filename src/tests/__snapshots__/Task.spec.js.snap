// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task completed should render the Task Component correctly 1`] = `
<tr
  className="task hover "
  id={0}
  key="0"
>
  <td>
    <input
      checked={true}
      className="mr-3  mb-3"
      id={0}
      ischecked="true"
      type="checkbox"
    />
  </td>
  <td
    className="strikeout"
  >
    <Link
      className="text-dark  mr-3 mb-3"
      id={0}
      onClick={[Function]}
      to="/view"
    >
      123
    </Link>
  </td>
  <td
    className="strikeout"
  >
    123
  </td>
  <td
    className="strikeout"
  >
    132
  </td>
  <td>
    <Link
      className="d-none"
      id="task_0"
      to="/edit"
    >
      <button
        className="btn btn-danger mr-3   d-none"
        id="task_0"
        onClick={[Function]}
      >
        Edit
      </button>
    </Link>
  </td>
  <td>
    <button
      className="btn btn-warning  d-none"
      id="task_0"
      onClick={[Function]}
    >
      Remove
    </button>
  </td>
  <td>
    <button
      className="btn btn-info"
      onClick={[Function]}
    />
  </td>
</tr>
`;

exports[`Task not completed should render the Task Component correctly 1`] = `
<tr
  className="task hover "
  id={0}
  key="0"
>
  <td>
    <input
      checked={true}
      className="mr-3  mb-3"
      id={0}
      ischecked="true"
      type="checkbox"
    />
  </td>
  <td>
    <Link
      className="text-dark  mr-3 mb-3"
      id={0}
      onClick={[Function]}
      to="/view"
    >
      123
    </Link>
  </td>
  <td>
    123
  </td>
  <td>
    132
  </td>
  <td>
    <Link
      className="d-none"
      id="task_0"
      to="/edit"
    >
      <button
        className="btn btn-danger mr-3   d-none"
        id="task_0"
        onClick={[Function]}
      >
        Edit
      </button>
    </Link>
  </td>
  <td>
    <button
      className="btn btn-warning  d-none"
      id="task_0"
      onClick={[Function]}
    >
      Remove
    </button>
  </td>
  <td>
    <button
      className="btn btn-info"
      onClick={[Function]}
    />
  </td>
</tr>
`;
